fiftyone/core/odm/dataset.py

```diff
@@ -315,6 +316,137 @@ def _id(self):
     def _id(self, value):
         self.id = str(value)
 
+    def _validate(self):
```

Comment:
@lanzhenw I changed this to a private `_validate()` method because if we were to implement it as `clean()`, it would cause it to become impossible to load an existing dataset that happens to have an invalid color scheme.

The only backwards compatible thing I can think of to do is to tell users to call `dataset.app_config.color_scheme._validate()` if they want to know if their color scheme is valid or not.